lib/update/phase_init.go

```diff
@@ -369,7 +386,30 @@ func removeLegacyUpdateDirectory(log log.FieldLogger) error {
 		return nil
 	}
 
-	log.Debugf("removing legacy update dictory %v", updateDir)
+	log.Debugf("Removing legacy update dictory %v.", updateDir)
 	err = os.RemoveAll(updateDir)
 	return trace.ConvertSystemError(err)
 }
+
+// Rollback rolls back the init phase
+func (p *updatePhaseInit) Rollback(context.Context) error {
+	if err := p.removeConfiguredPackages(); err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+// removeConfiguredPackages removes packages configured during init phase
+func (p *updatePhaseInit) removeConfiguredPackages() error {
```

Comment:
I wonder if this would need to be done after the update regardless - i.e. removing the auxiliary labels?

Reply:
Hm, why? I think it's okay to keep configured packages in the cluster db (e.g. if you need to re-pull it locally for whatever reason or what not).